go/libraries/doltcore/sqle/dtables/diff_table.go

```diff
@@ -395,6 +395,11 @@ func (dp *diffPartition) isDiffablePartition(ctx *sql.Context) (bool, error) {
 		return false, err
 	}
 
+	// dp.to can be nil if a table was deleted and then recreated
+	if dp.to == nil {
```

Comment:
Same comment, is this bug related to this PR>

Reply:
This one is related to the functionality of DOLT_BLAME (for some cases at least), since it's built on top of DOLT_DIFF_<TABLE>. It also felt like it needed a little bit more thorough vetting to make sure this was the right change though.

Reply:
To keep this PR simple... I went ahead and reverted this commit, and pushed it onto a new local branch (fulghum/diff-table-bugfix-2496). I'll follow up on that one after landing the DOLT_BLAME system view.